### PR TITLE
[BEAM-4461] Resubmit PR to switch SQL over to schema transfomr

### PR DESCRIPTION
**Pull request by reuvenlax**
The previous PR failed because of a missing override for getAccumulatorCoder. Since the override was missing the default delegated to CoderRegistry, which failed due to type erasure..

### CODE REVIEW

**Comment by lgajowy** [comment]
Run Java PreCommit

**Comment by lgajowy** [comment]
Run Java PreCommit

**Comment by lgajowy** [comment]
It looks good to me. Please "squash and merge" it if all checks are green.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java [hunk 1]
```diff
@@ -17,77 +17,146 @@
...
       AggregateCall call, Schema.Field field, String functionName) {
+    CombineFn combineFn;
     if (call.getAggregation() instanceof SqlUserDefinedAggFunction) {
-      return getUdafCombineFn(call);
+      combineFn = getUdafCombineFn(call);
+    } else {
+      combineFn = BeamBuiltinAggregations.create(functionName, field.getType().getTypeName());
+    }
+    if (call.getArgList().isEmpty()) {
+      return new SingleInputCombiner(combineFn);
```
Comment (akedin) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java]: `ConstantEmpty` here?

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamAggregationRel.java [hunk 2]
```diff
@@ -118,58 +114,51 @@ public RelWriter explainTerms(RelWriter pw) {
...
 
     return new Transform(
-        windowFn, windowFieldIndex, inputSchema, getGroupSet(), aggregationAdapters, outputSchema);
+        windowFn, windowFieldIndex, getGroupSet(), aggregationAdapters, outputSchema);
+  }
+
+  private static class FieldAggregation implements Serializable {
+    final List<Integer> inputs;
+    final CombineFn combineFn;
+    final Field outputField;
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamAggregationRel.java]: AggregationCombineFnAdapter is simply a factory that returns (possibly multiple) combiners. This is the only container class.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java [hunk 3]
```diff
@@ -17,77 +17,146 @@
...
-    this.argsAdapter = argsAdapter;
+  private static class MultiInputCombiner extends WrappedCombinerBase<Row> {
+    MultiInputCombiner(CombineFn<Row, Object, Object> combineFn) {
+      super(combineFn);
+    }
+
+    @Override
+    Row getInput(Row input) {
+      for (Object o : input.getValues()) {
+        if (o == null) {
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java]: correct, but the returned list might contain null values.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java (hunk 1 above)
Comment (reuvenlax) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java]: No, that is not the same semantics (different types)

**Pull request merged**